### PR TITLE
Small Reader Improvements: Add Related Posts

### PR DESCRIPTION
**Pull request by momo-ozawa**
Part of #15942
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/360

## Description
- Added related posts to the bottom of the reader detail screen 
- TODO: handle cell selection for related posts (Will make a separate PR)

## How to test
1. Go to reader tab
2. Tap on a reader post and open post details
3. Scroll to the bottom of the post
4. ✅ The related posts section should be displayed (Up to 2 "More from {blog}" posts / exactly 2 "More on WordPress.com" posts) 
5. ⚠️ Tapping the cells won't do anything for now

## Screenshot

Design ref: I5jpMcL84tYRh3FQJZs0fF-fi-315%3A89

Simulator Screen Shot - iPhone 12 Pro - 2021-03-03 at 17 51 43| Simulator Screen Shot - iPhone 12 Pro - 2021-03-03 at 17 51 52
--- | ---

## PR submission checklist:

- [ ] I have considered adding unit tests where possible.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift [hunk 1]
```diff
@@ -455,6 +491,66 @@ extension ReaderDetailViewController: StoryboardLoadable {
...
+
+    private func getSectionTitle(for section: Int) -> String? {
+        switch RemoteReaderSimplePost.PostType(rawValue: section) {
+        case .local:
+            guard let blogName = post?.blogNameForDisplay() else {
+                return nil
+            }
+            return String(format: Strings.localPostsSectionTitle, blogName)
+        case .global:
+            return Strings.globallPostsSectionTitle
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift]:
```suggestion
            return Strings.globalPostsSectionTitle
```

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift [hunk 2]
```diff
@@ -644,6 +734,19 @@ extension ReaderDetailViewController: UIViewControllerRestoration {
...
+// MARK: - Strings
+extension ReaderDetailViewController {
+
+    private struct Strings {
+        static let backButtonAccessibilityLabel = NSLocalizedString("Back", comment: "Spoken accessibility label")
+        static let safariButtonAccessibilityLabel = NSLocalizedString("Open in Safari", comment: "Spoken accessibility label")
+        static let shareButtonAccessibilityLabel = NSLocalizedString("Share", comment: "Spoken accessibility label")
+        static let moreButtonAccessibilityLabel = NSLocalizedString("More", comment: "Spoken accessibility label")
+        static let localPostsSectionTitle = NSLocalizedString("More from %1$@", comment: "Section title for local related posts. %1$@ is a placeholder for the blog display name.")
+        static let globallPostsSectionTitle = NSLocalizedString("More on WordPress.com", comment: "Section title for global related posts.")
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift]:
```suggestion
        static let globalPostsSectionTitle = NSLocalizedString("More on WordPress.com", comment: "Section title for global related posts.")
```

File: WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderRelatedPostsCell.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            featuredImageView.isHidden = true
+            return
+        }
+
+        featuredImageView.downloadImage(from: featuredImageUrl)
+    }
+
+    private func configureLabels(for post: RemoteReaderSimplePost) {
+        titleLabel.text = NSAttributedString.attributedStringWithHTML(post.title, attributes: nil).string
+        excerptLabel.text = NSAttributedString.attributedStringWithHTML(post.excerpt, attributes: nil).string
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderRelatedPostsCell.swift]: How do you feel about reusing the `makePlainText` method that is used to generate the summary on other areas in the reader?

File: WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderRelatedPostsCell.swift [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+              let featuredImageUrl = URL(string: urlString) else {
+            featuredImageView.isHidden = true
+            return
+        }
+
+        featuredImageView.downloadImage(from: featuredImageUrl)
+    }
+
+    private func configureLabels(for post: RemoteReaderSimplePost) {
+        titleLabel.text = NSAttributedString.attributedStringWithHTML(post.title, attributes: nil).string
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderRelatedPostsCell.swift]: See comment below.

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Nice work! I left a few comments below:

When testing RTL it looks like the section header doesn't adjust its position. 


cc @mbshakti 
Should we bold or highlight the section headers, I find they blend in with the content and are kind of hard to see.

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift (hunk 2 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift]: Fixed! 7d9d5d9 🙈

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift (hunk 1 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift]: Fixed! 7d9d5d9 🙈

File: WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderRelatedPostsCell.swift (hunk 3 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderRelatedPostsCell.swift]: Thanks for the tip!
Fixed! 7b2a521

File: WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderRelatedPostsCell.swift (hunk 4 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderRelatedPostsCell.swift]: Fixed! 7b2a521

**Comment by momo-ozawa** [comment]
@emilylaguna 
> When testing RTL it looks like the section header doesn't adjust its position.

Fixed! 617e44f

**Review by emilylaguna — APPROVED** [review]
🚢 We can address the UI updates later if needed.

**Comment by mbshakti** [comment]
> Should we bold or highlight the section headers, I find they blend in with the content and are kind of hard to see.

I was actually thinking the same (that it blends in with the content). The section header is not that important, the actual content is more important, so maybe we can try using a smaller type style. I think it's `subheadline` right now, could we use `footnote` instead? 

otherwise is there a build for this? would love to play around with it 🙏

**Comment by emilylaguna** [comment]
> otherwise is there a build for this? would love to play around with it 🙏

@mbshakti the build is `43385` Keep in mind the cells don't go anywhere yet.

**Comment by momo-ozawa** [comment]
@mbshakti (cc @emilylaguna)
I've updated the section title font style to footnote 😄 
The new build number is`43429`

**Pull request merged**